### ISSUE TITLE
syslog flooded with multicastBeacon: Failed service / sendmsg: network is unreachable (Syncthing 1.12.1)

### ISSUE THREAD

**Issue by mackonsti** [syncthing/syncthing#7300]
### Does your log mention database corruption?
No, no trace of this in the logs.

### Include required information
Not sure if this is an issue with latest Syncthing v1.12.1 release, but I have had Syncthing installed for months and it's the first time the **syslog** is flooded with these messages; the home router has for 22000 TCP open just fine, redirecting traffic to the NAS IP, too. The NAS itself has no firewall set up.

This is a sample of the entries in syslog:

```
Jan 24 14:09:28 OMV-NAS syncthing[4689]: 2021/01/24 14:09:28 multicastBeacon: Exiting backoff state.
Jan 24 14:09:28 OMV-NAS syncthing[4689]: 2021/01/24 14:09:28 multicastBeacon: Failed service 'Service@0xc000030c00 created by multicastBeacon@0xc0000ac5a0/writer' (1.000000 failures of 2.000000), restarting: true, error: write udp [::]:35186->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:09:28 OMV-NAS syncthing[4689]: 2021/01/24 14:09:28 multicastBeacon: Failed service 'Service@0xc000030c00 created by multicastBeacon@0xc0000ac5a0/writer' (1.999972 failures of 2.000000), restarting: true, error: write udp [::]:33708->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:09:48 OMV-NAS syncthing[4689]: 2021/01/24 14:09:48 multicastBeacon: Entering the backoff state.
Jan 24 14:09:48 OMV-NAS syncthing[4689]: 2021/01/24 14:09:48 multicastBeacon: Failed service 'Service@0xc000030c00 created by multicastBeacon@0xc0000ac5a0/writer' (2.260529 failures of 2.000000), restarting: false, error: write udp [::]:37499->[ff12::8384]:21027: sendmsg: network is unreachable
```
Please be sure to include at least:

- which version of Syncthing and what operating system you are using:
**Version 1.12.1**

- browser and version, if applicable:
 **Not applicable** ; this ticket refers to a running service on OpenMediaVault 4.x NAS (Intel/Debian). The NAS has no firewall service running, as far as I can tell; only the home router (OpenWRT) redirects incoming traffic:
WAN:TCP 22000 -> NAS:TCP 22000
WAN:UDP 21027 -> NAS:UDP 21027
Do I need the port 21027 open on my router? I added it as soon as I discovered these errors in **syslog** but the error persists...

- what happened:
Recently updated to syncthing v1.12.0 "Fermium Flea" and then to v1.12.1 as instructed and noticed these on the NAS own syslog entries ; no changes done on the home router ; the client(s) most notably MacOS were also updated to this v1.12.1 too.

- what you expected to happen instead, and
I expect to not get such errors as the **port forwarding** seems properly set up. I do not understand where this error comes from; I added UDP 21027 on the home router (OpenWRT) too, but error persists in **syslog** of NAS. I followed the firewall instructions from https://docs.syncthing.net/users/firewall.html

- any steps to reproduce the problem.
Restart server or Syncthing service; monitor **syslog** entries. They are flooded with this error (sensitive data are masked with XXXX):

```
Jan 24 14:33:50 OMV-NAS systemd[1]: Started Syncthing - Open Source Continuous File Synchronization for user XXXX.
Jan 24 14:33:50 OMV-NAS syncthing[6654]: [start] INFO: syncthing v1.12.1 "Fermium Flea" (go1.15.6 linux-amd64) deb@build.syncthing.net 2020-12-06 12:46:27 UTC [noupgrade]
Jan 24 14:33:50 OMV-NAS syncthing[6654]: [start] INFO: Using large-database tuning
Jan 24 14:33:51 OMV-NAS syncthing[6654]: [OLP3I] INFO: My ID: XXXX
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Single thread SHA256 performance is 202 MB/s using minio/sha256-simd (197 MB/s using crypto/sha256).
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Hashing performance is 170.94 MB/s
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Overall send rate is unlimited, receive rate is unlimited
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Using discovery mechanism: global discovery server https://discovery.syncthing.net/v2/?noannounce&id=XXXX
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Using discovery mechanism: global discovery server https://discovery-v4.syncthing.net/v2/?nolookup&id=XXXX
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Using discovery mechanism: global discovery server https://discovery-v6.syncthing.net/v2/?nolookup&id=XXXX
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Using discovery mechanism: IPv4 local broadcast discovery on port 21027
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Using discovery mechanism: IPv6 local multicast discovery on address [ff12::8384]:21027
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Ready to synchronize "XXXX" (sendreceive)
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: TCP listener ([::]:22000) starting
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Ready to synchronize "XXXX" (sendreceive)
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Ready to synchronize "XXXX" (sendreceive)
Jan 24 14:33:52 OMV-NAS syncthing[6654]: 2021/01/24 14:33:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.000000 failures of 2.000000), restarting: true, error: write udp [::]:49228->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: GUI and API listening on [::]:8384
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Access the GUI via the following URL: http://127.0.0.1:8384/
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: My name is "XXXX"
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Device XXXX is "MacBookPro" at [dynamic]
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Device XXXX is "MacMini" at [dynamic]
Jan 24 14:33:52 OMV-NAS syncthing[6654]: [OLP3I] INFO: Completed initial scan of sendreceive folder "XXXX"
Jan 24 14:33:58 OMV-NAS syncthing[6654]: [OLP3I] INFO: Completed initial scan of sendreceive folder "XXXX"
Jan 24 14:34:03 OMV-NAS syncthing[6654]: [OLP3I] INFO: Completed initial scan of sendreceive folder "XXXX"
Jan 24 14:34:06 OMV-NAS syncthing[6654]: [OLP3I] INFO: Detected 1 NAT service
Jan 24 14:34:22 OMV-NAS syncthing[6654]: 2021/01/24 14:34:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.500037 failures of 2.000000), restarting: true, error: write udp [::]:54963->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:34:46 OMV-NAS syncthing[6654]: [OLP3I] INFO: Established secure connection to XXXX at 192.168.X.X:22000-192.168.X.X:57759/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256
Jan 24 14:34:46 OMV-NAS syncthing[6654]: [OLP3I] INFO: Device XXXX client is "syncthing v1.12.1" named "MacMini" at 192.168.X.X:22000-192.168.X.X:57759/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256
Jan 24 14:34:52 OMV-NAS syncthing[6654]: 2021/01/24 14:34:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.750008 failures of 2.000000), restarting: true, error: write udp [::]:41424->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:35:22 OMV-NAS syncthing[6654]: 2021/01/24 14:35:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.875012 failures of 2.000000), restarting: true, error: write udp [::]:46035->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:35:52 OMV-NAS syncthing[6654]: 2021/01/24 14:35:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.937507 failures of 2.000000), restarting: true, error: write udp [::]:54853->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:36:22 OMV-NAS syncthing[6654]: 2021/01/24 14:36:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.968752 failures of 2.000000), restarting: true, error: write udp [::]:57914->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:36:52 OMV-NAS syncthing[6654]: 2021/01/24 14:36:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.984380 failures of 2.000000), restarting: true, error: write udp [::]:39832->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:37:22 OMV-NAS syncthing[6654]: 2021/01/24 14:37:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.992189 failures of 2.000000), restarting: true, error: write udp [::]:44526->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:37:52 OMV-NAS syncthing[6654]: 2021/01/24 14:37:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.996086 failures of 2.000000), restarting: true, error: write udp [::]:40780->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:38:22 OMV-NAS syncthing[6654]: 2021/01/24 14:38:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.998041 failures of 2.000000), restarting: true, error: write udp [::]:34204->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:38:52 OMV-NAS syncthing[6654]: 2021/01/24 14:38:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999031 failures of 2.000000), restarting: true, error: write udp [::]:53810->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:39:22 OMV-NAS syncthing[6654]: 2021/01/24 14:39:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999516 failures of 2.000000), restarting: true, error: write udp [::]:44562->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:39:52 OMV-NAS syncthing[6654]: 2021/01/24 14:39:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999758 failures of 2.000000), restarting: true, error: write udp [::]:46878->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:40:22 OMV-NAS syncthing[6654]: 2021/01/24 14:40:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999881 failures of 2.000000), restarting: true, error: write udp [::]:45036->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:40:52 OMV-NAS syncthing[6654]: 2021/01/24 14:40:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999937 failures of 2.000000), restarting: true, error: write udp [::]:47082->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:41:22 OMV-NAS syncthing[6654]: 2021/01/24 14:41:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999955 failures of 2.000000), restarting: true, error: write udp [::]:44611->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:41:52 OMV-NAS syncthing[6654]: 2021/01/24 14:41:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999982 failures of 2.000000), restarting: true, error: write udp [::]:43096->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:42:22 OMV-NAS syncthing[6654]: 2021/01/24 14:42:22 multicastBeacon: Entering the backoff state.
Jan 24 14:42:22 OMV-NAS syncthing[6654]: 2021/01/24 14:42:22 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (2.000000 failures of 2.000000), restarting: false, error: write udp [::]:50772->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:43:27 OMV-NAS syncthing[6654]: 2021/01/24 14:43:27 multicastBeacon: Exiting backoff state.
Jan 24 14:43:27 OMV-NAS syncthing[6654]: 2021/01/24 14:43:27 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.000000 failures of 2.000000), restarting: true, error: write udp [::]:48914->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:43:27 OMV-NAS syncthing[6654]: 2021/01/24 14:43:27 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (1.999969 failures of 2.000000), restarting: true, error: write udp [::]:59694->[ff12::8384]:21027: sendmsg: network is unreachable
Jan 24 14:43:52 OMV-NAS syncthing[6654]: 2021/01/24 14:43:52 multicastBeacon: Entering the backoff state.
Jan 24 14:43:52 OMV-NAS syncthing[6654]: 2021/01/24 14:43:52 multicastBeacon: Failed service 'Service@0xc000191380 created by multicastBeacon@0xc00019ca00/writer' (2.133120 failures of 2.000000), restarting: false, error: write udp [::]:60281->[ff12::8384]:21027: sendmsg: network is unreachable
```

This is the configuration part of my `config.xml` on the NAS (removed the folder and devices and masked sensitive data with XXX) which is really the default after installation:

```
    <options>
        <listenAddress>tcp://0.0.0.0:22000</listenAddress>
        <globalAnnounceServer>default</globalAnnounceServer>
        <globalAnnounceEnabled>true</globalAnnounceEnabled>
        <localAnnounceEnabled>true</localAnnounceEnabled>
        <localAnnouncePort>21027</localAnnouncePort>
        <localAnnounceMCAddr>[XXXX::XXXX]:21027</localAnnounceMCAddr>
        <maxSendKbps>0</maxSendKbps>
        <maxRecvKbps>0</maxRecvKbps>
        <reconnectionIntervalS>60</reconnectionIntervalS>
        <relaysEnabled>true</relaysEnabled>
        <relayReconnectIntervalM>10</relayReconnectIntervalM>
        <startBrowser>false</startBrowser>
        <natEnabled>true</natEnabled>
        <natLeaseMinutes>60</natLeaseMinutes>
        <natRenewalMinutes>30</natRenewalMinutes>
        <natTimeoutSeconds>10</natTimeoutSeconds>
        <urAccepted>3</urAccepted>
        <urSeen>3</urSeen>
        <urUniqueID>XXXX</urUniqueID>
        <urURL>https://data.syncthing.net/newdata</urURL>
        <urPostInsecurely>false</urPostInsecurely>
        <urInitialDelayS>1800</urInitialDelayS>
        <restartOnWakeup>true</restartOnWakeup>
        <autoUpgradeIntervalH>12</autoUpgradeIntervalH>
        <upgradeToPreReleases>false</upgradeToPreReleases>
        <keepTemporariesH>24</keepTemporariesH>
        <cacheIgnoredFiles>false</cacheIgnoredFiles>
        <progressUpdateIntervalS>5</progressUpdateIntervalS>
        <limitBandwidthInLan>false</limitBandwidthInLan>
        <minHomeDiskFree unit="%">1</minHomeDiskFree>
        <releasesURL>https://upgrades.syncthing.net/meta.json</releasesURL>
        <overwriteRemoteDeviceNamesOnConnect>false</overwriteRemoteDeviceNamesOnConnect>
        <tempIndexMinBlocks>10</tempIndexMinBlocks>
        <trafficClass>0</trafficClass>
        <defaultFolderPath>/home/XXXX</defaultFolderPath>
        <setLowPriority>true</setLowPriority>
        <maxFolderConcurrency>0</maxFolderConcurrency>
        <crashReportingURL>https://crash.syncthing.net/newcrash</crashReportingURL>
        <crashReportingEnabled>true</crashReportingEnabled>
        <stunKeepaliveStartS>180</stunKeepaliveStartS>
        <stunKeepaliveMinS>20</stunKeepaliveMinS>
        <stunServer>default</stunServer>
        <databaseTuning>auto</databaseTuning>
        <maxConcurrentIncomingRequestKiB>0</maxConcurrentIncomingRequestKiB>
        <announceLANAddresses>true</announceLANAddresses>
    </options>
```

Any help and insight is very welcome. Thank you.

**Comment by calmh** [comment]
> write udp [::]:57914->[ff12::8384]:21027: sendmsg: network is unreachable

It seems IPv6 is unavailable or disabled. I don't think you can disable local discovery over IPv6 specifically, so I think either enable IPv6, disable local discovery, or live with (or filter out) the complaints...

**Comment by mackonsti** [comment]
Thanks @calmh for replying.

1. When you say IPv6 is "unavailable", are you referring to the NAS server that acts as Syncthing server not also getting an IPv6 assigned? Or the connection _to_ the discovery servers? Indeed, on OpenWRT I have disabled DHCP in IPv6.

All this time, I have had my NAS with a fixed IP in a home IPv4 setup; so not sure how to setup IPv6 so Syncthing doesn't complain... I admit I a not comfortable fiddling with IPv6.

2. In the main Syncthing server page, I read **Discovery | 2/3** ("click to see discovery failures") and I see this popping up:
```
global@https://discovery-v6.syncthing.net/v2/: Post "https://discovery-v6.syncthing.net/v2/":
dial tcp [2400:6180:100:d0::741:a001]:443: connect: network is unreachable
Failure to connect to IPv6 servers is expected if there is no IPv6 connectivity.
```

In the Settings > **Connections** tab, I tried forcing to IPv4 with the following entry (this way my office laptop talks to this Syncthing server via DuckDNS reference):
```
Sync Protocol Listen Addresses
tcp4://0.0.0.0:22000, dynamic+https://relays.syncthing.net/endpoint
``` 
Would a `tcp4` reference _not_ force IPv4 including discovery?

3. In Advanced Settings > **Options** tab, I see a reference to:
```
Local Announce Enabled = Disabled (now)
Local Announce MC Addr = [ff12::8384]:21027
Local Announce Port = 21027
```
...not sure if this has any impact changing this or removing it?

4. Any idea if we eventually need to open 21027 port on the router (and redirect to the Syncthing server) per instructions from https://docs.syncthing.net/users/firewall.html ? This is done today in an IPv4 manner.

**NOTE:** OK so I _did_ disable Local Discovery in Syncthing server settings... the above error in syslog is not appearing (that frequently) but not sure of the impact of clients, time will tell.

Thanks again.

**Comment by calmh** [comment]
There are no plans to disable IPv6 specifically.

**Comment by mackonsti** [comment]
Please @calmh can you at least kindly advise _how_ to filter these out from **syslog**? It's growing very fast and consumes resources that could otherwise be saved... many thanks.... Global or Local Discovery results to these IPv6 errors, and it would be great at least to minimise them to the server logs as much as possible... Thanks again.

**Comment by calmh** [comment]
I have no idea, that's between you and your syslog implementation. If you're starting Syncthing with a script then a simple grep on the output should suffice.

**Comment by xueshuaihang** [comment]
I have the same problem, I am very upset because of this problem, this problem happened in CentOS 7.No problem on Windows.I don't quite understand what you said above, please let me know if there is any solution.

**Comment by imsodin** [comment]
This lines will be logged at debug level in v1.13.0 (thus is already fixed in the current RC: v1.13.0-rc.1), see https://github.com/syncthing/syncthing/pull/7202

If you need any help dealing with logs, please post to https://forum.syncthing.net

**Comment by xueshuaihang** [comment]
> This lines will be logged at debug level in v1.13.0 (thus is already fixed in the current RC: v1.13.0-rc.1), see #7202
> 
> If you need any help dealing with logs, please post to https://forum.syncthing.net

Thank you very much for the information you provided to me, which made me no longer confused
